### PR TITLE
Upgrade to HCL 2.0

### PR DESCRIPTION
**Pull request by ksatirli**
This is a re-structuring of the module with HCL 2.0 rewrites and a lot of clean-ups.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  - [Dependencies](#dependencies)
+  - [Usage](#usage)
+    - [Module Variables](#module-variables)
+    - [Module Outputs](#module-outputs)
+  - [Author Information](#author-information)
+  - [License](#license)
+
+## Requirements
+
+This module requires Terraform version `0.12.0` or newer.
```
Comment (kibertoad) [inline README.md:19]: `required_version = "> 0.12.0"`

Is this correct?

Comment (ksatirli) [reply README.md:19]: Great catch! Fixed this in https://github.com/operatehappy/terraform-module-aws-route53-workmail-records/pull/1/commits/c9c4b20666f47a34ecd8a0756829e511925f6484

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+fail_fast: true
+repos:
+  -
+    repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.0.0
```
Comment (kibertoad) [inline .pre-commit-config.yaml]: Seems like 2.3.0 is available

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: check-json
+      - id: check-yaml
+      - id: detect-aws-credentials
+        args: ["--allow-missing-credentials"]
+      - id: check-vcs-permalinks
+  -
+    repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 1.10.2
```
Comment (kibertoad) [inline .pre-commit-config.yaml]: Latest is 1.11.0

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    hooks:
+      - id: require-ascii
+      - id: shellcheck
+
+  -
+    repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.13.0
```
Comment (kibertoad) [inline .pre-commit-config.yaml]: Latest is 0.18.0, not sure how much they did break

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    hooks:
+      -
+        id: markdownlint
+        args: ["--config=.mdlrc"]
+
+  -
+    repo: https://github.com/antonbabenko/pre-commit-terraform
+    rev: v1.7.3
```
Comment (kibertoad) [inline .pre-commit-config.yaml]: Latest is 1.19.0

File: .release-it.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
+	"dry-run": true,
```
Comment (kibertoad) [inline .release-it.json]: dry-run is supposed to be overriden manually?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (ksatirli) [reply .pre-commit-config.yaml]: Good catch, fixed!

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ksatirli) [reply .pre-commit-config.yaml]: Good catch, fixed!

File: .pre-commit-config.yaml (hunk 5 above)
Comment (ksatirli) [reply .pre-commit-config.yaml]: Good catch, fixed!

File: .pre-commit-config.yaml (hunk 4 above)
Comment (ksatirli) [reply .pre-commit-config.yaml]: Good catch, fixed!

**Review by kibertoad — APPROVED** [review]
Perfection.

File: .release-it.json (hunk 6 above)
Comment (ksatirli) [reply .release-it.json]: Great catch - this was indeed an oversight and used for testing.

**Pull request merged**